File: if문/2884.py
h, m = map(int, input().split())
if (m >= 45):
    print(h, m-45)
elif (m < 45 and 1 <= h <= 23):
    print(h-1, m+15)
elif (m < 45 and h == 0):
    print(23, m+15)

# 처음 쓴 답안(오답)
# if (1 <= h <= 23 and m >= 45):
#     print(h, m-45)
# elif (1 <= h <= 23 and m < 45):
#     print(h-1, m+15)
# elif (h == 0 and m >= 45):
#     print(23, m-45)
# elif (h == 0 and m < 45):
#     print(23, m+15)
# 오답인 이유: h==0 and m>=45인 세번째 elif의 경우 오류

File: if문/2884.js
// 알람 시계

const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

// let input = [];

rl.on("line", (line) => {
    let input = line.split(" ");
    let H = Number(input[0]);
    let M = Number(input[1]);

    if (M - 45 < 0) {
        M += 15;
        H -= 1;

        if (H === -1) {
            H = 23;
        }
    } else {
        M -= 45;
    }
    console.log(H + " " + M);
}).on("close", () => {
    process.exit();
})
